src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -315,13 +315,43 @@ public static Type GetUnderlyingType(Type enumType)
             (TEnum[])GetValues(typeof(TEnum));
 #endif
 
-        [RequiresDynamicCode("It might not be possible to create an array of the enum type at runtime. Use the GetValues<TEnum> overload instead.")]
+        [RequiresDynamicCode("It might not be possible to create an array of the enum type at runtime. Use the GetValues<TEnum> overload or the GetValuesAsUnderlyingType method instead.")]
         public static Array GetValues(Type enumType)
         {
             ArgumentNullException.ThrowIfNull(enumType);
             return enumType.GetEnumValues();
         }
 
+        /// <summary>
+        /// Gets an array of values of the underlying type of the Enum.
+        /// </summary>
+        /// <typeparam name="TEnum">Enum type</typeparam>
+        /// /// <remarks>
+        /// This method can be used to get enum values when creating an Array of Enum is challenging
+        /// For example, reflection-only context or on a platform where runtime codegen is not available.
```

Comment:
reflection-only context -> System.Reflection.MetadataLoadContext

reflection-only context was a thing in .NET Framework. It has been deprecated in .NET Core.